internal/setup/setup.go

```diff
@@ -0,0 +1,195 @@
+	separator := "————————————————————————————————————————————————————————————————————————————————"
+	fmt.Println(separator)
+	fmt.Print("🛠️ Let's first set up your configuration and catalog repo...\n\n")
+	// Determine catalog directory
+	if catalogDir == "" {
+		// Try loading catalog dir from config file to use as prompt default value
+		cfg, err := config.Load(configDir, catalogDir)
+		if err == nil {
+			catalogDir = cfg.CatalogDir
+		} else {
+			catalogDir = defaultCatalogDir
+		}
+
+		// Prompt user for catalog directory using survey (defaults to $HOME/.joy)
+		err = survey.AskOne(&survey.Input{
+			Message: "🎯 Where does (or should) your local catalog reside?",
+			Help:    "This is where we will clone your catalog repo, but only if it's not already there.",
+			Default: catalogDir,
+		},
+			&catalogDir,
+			survey.WithValidator(survey.Required),
+		)
+		if err != nil {
+			return fmt.Errorf("prompting for catalog directory: %w", err)
+		}
+	}
+
+	// Expand tilde to home directory
+	homePrefix := "~/"
+	if strings.HasPrefix(catalogDir, homePrefix) {
+		homeDir, err := os.UserHomeDir()
+		if err != nil {
+			return fmt.Errorf("getting home directory: %w", err)
+		}
+		catalogDir = path.Join(homeDir, strings.TrimPrefix(catalogDir, homePrefix))
+	}
+
+	// Check if catalog directory exists
+	if _, err := os.Stat(catalogDir); err != nil {
+		if os.IsNotExist(err) {
+			shouldClone := true
+			if catalogRepo == "" {
+				// Prompt user whether to clone it
+				err := survey.AskOne(&survey.Confirm{
+					Message: "🤷 No trace of catalog at given location, clone it?",
+					Default: true,
+				},
+					&shouldClone,
+				)
+				if err != nil {
+					return fmt.Errorf("prompting for catalog cloning: %w", err)
+				}
+
+				// Prompt user for catalog repo
+				err = survey.AskOne(&survey.Input{
+					Message: "📦 What's your catalog repo URL?",
+				},
+					&catalogRepo,
+					survey.WithValidator(survey.Required),
+				)
+				if err != nil {
+					return fmt.Errorf("prompting for catalog repo: %w", err)
+				}
+			}
+
+			// Clone catalog
+			if shouldClone {
+				cmd := exec.Command("git", "clone", catalogRepo, catalogDir)
+				output, err := cmd.CombinedOutput()
+				if err != nil {
+					return fmt.Errorf("cloning catalog: %s", output)
+				}
+				fmt.Printf("✅ Cloned catalog from %s to %s\n", style.Link(catalogRepo), style.Code(catalogDir))
+			} else {
+				fmt.Println("😬 Sorry, cannot continue without catalog!")
+				os.Exit(1)
+			}
+
+		} else {
+			return fmt.Errorf("checking for catalog directory %s: %w", catalogDir, err)
+		}
+	}
+
+	// Check catalog directory
+	cat, err := catalog.Load(catalog.LoadOpts{
+		Dir:          catalogDir,
+		LoadEnvs:     true,
+		LoadProjects: true,
+		LoadReleases: true,
+		ResolveRefs:  true,
+	})
+	if err != nil {
+		fmt.Printf("🤯 Whoa! Found the catalog, but it's speaking gibberish. Check this error and try again:\n%v\n", err)
```

Comment:
In this context, how will retrying will help with the issue?

I think we could use a way to show the user more help but keep it private. Basically, we have to do extra work since this is public repository.


```bash
➜  catalog git:(master) ✗ cat configuration.yaml
---
peopleRepository: git@github.com:nestoca/people
extraErrorMessage: Please refer to <aPrivatePageInConfluence> or ask help in #our-private-help-slack-channel
```